sphinxcontrib/autoclassdiag.py

```diff
@@ -1,57 +1,55 @@
-"""
-generate mermaid code that represent the inheritance of classes
-defined in a given module
-
-Check https://github.com/mgaitan/sphinxcontrib-mermaid/issues/5
-
-Original code by Zulko: https://gist.github.com/Zulko/e0910cac1b27bcc3a1e6585eaee60121
-
-"""
-from __future__ import print_function
 import inspect
 from sphinx.util import import_object
 
-def class_name(cls):
-    """Return a string representing the class"""
-    # NOTE: can be changed to str(class) for more complete class info
-    return cls.__name__
 
 
-class ClassDiagram(object):
+def class_diagram(*cls_names, full=False):
+    inheritances = set()
 
-    def __init__(self, module_path, base_module=None):
+    def get_tree(cls):
+        for base in cls.__bases__:
+            if base.__name__ == 'object':
+                continue
+            inheritances.add((base.__name__, cls.__name__))
+            if full:
+                get_tree(base)
 
-        self.module = import_object(module_path)
-        self.base_module = base_module or self.module.__name__
-        self.module_classes = set()
-        self.inheritances = []
-        self._populate_tree()
+    for cls_name in cls_names:
 
-    def _inspect_class(self, cls):
+        cls = import_object(cls_name)
         if not inspect.isclass(cls):
```

Comment:
This does not work for modules anymore.

It was a useful feature, because I used to specify only a module name and all classes there magically appeared.

Reply:
now modules are accepted, with a couple of options to fine-grain what is included.